driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -1259,6 +1259,19 @@ private void validateProxyParameters() {
             throw new IllegalArgumentException(
                     "Both proxyUsername and proxyPassword must be set together. They cannot be set individually");
         }
+
+        if (proxyHost != null && optionsMap.get("proxyhost").size() > 1) {
```

Comment:
`proxyHost != null` is used here (and similarly below) as a way to check that `optionsMap.get("proxyhost")` is not `null`. Let's not do that, because it is not only non-obvious, but also fragile. We may introduce a `static` method for checking that a `Collection` has more than one element, which checks for `null`s, instead of what we do now.

Reply:
Good shout - updated.